### PR TITLE
env-util: add unsetenv_erase() helper

### PR DESCRIPTION
**Pull request by poettering**
Let's unify how we remove secrets from the env block.

### CODE REVIEW

File: src/basic/env-util.c [hunk 1]
```diff
@@ -870,3 +870,18 @@ int getenv_path_list(const char *name, char ***ret_paths) {
         *ret_paths = TAKE_PTR(l);
         return 1;
 }
+
+int unsetenv_erase(const char *name) {
+        char *p;
+
```
Comment (bluca) [inline src/basic/env-util.c:876]: getenv() doesn't like NULL input, assert/check on `name`?

File: src/cryptenroll/cryptenroll-password.c [hunk 2]
```diff
@@ -27,8 +28,7 @@ int enroll_password(
                 if (!new_password)
                         return log_oom();
 
-                string_erase(e);
```
Comment (bluca) [inline src/cryptenroll/cryptenroll-password.c:30]: I don't follow, we don't need to do the erase thing anymore?

Comment (bluca) [reply src/cryptenroll/cryptenroll-password.c:30]: ie, the content of "e" is now left behind?

Comment (medhefgo) [reply src/cryptenroll/cryptenroll-password.c:30]: That's the whole point of this PR. e == getenv("NEWPASSWORD") here, so it'll get erased in unsetenv_erase. Afaik, getenv() points to some global area.

Comment (bluca) [reply src/cryptenroll/cryptenroll-password.c:30]: ah I missed that it was all globals

**Review by bluca — APPROVED** [review]
lgtm, just one nit - and maybe adding a unit test to `src/test/test-env-util.c` would be nice as well

**Comment by poettering** [comment]
pushed a new version. made the suggested change and added a test. since minor stuff updating the green label.

**Pull request merged**